socket_tls: add TLS 1.3 support using Mbed TLS (#77567)
zephyrproject-rtos/zephyr: doc/releases/migration-guide-4.0.rst

* The Kconfig options ``CONFIG_MBEDTLS_TLS_VERSION_1_0`` and ``CONFIG_MBEDTLS_TLS_VERSION_1_1``
  have been removed because Mbed TLS doesn't support TLS 1.0 and 1.1 anymore since v3.0. (:github:`76833`)

* The Kconfig option ``CONFIG_MBEDTLS_SSL_EXPORT_KEYS`` was removed because the

>   corresponding build symbol was removed from Mbed TLS since revision 3.1.0.

Reviewer: Rather either 
```suggestion
  corresponding build symbol was removed in Mbed TLS 3.1.0.
```
or
```suggestion
  corresponding build symbol doesn't exist anymore since Mbed TLS 3.1.0.
```
and up to you whether you want to add `version`.

>   ``MBEDTLS_SSL_EXPORT_KEYS`` is assumed to be default on in Mbed TLS.

Reviewer: I think this could be rephrased to be more clear. To me this sounds like the macro may still exist in Mbed TLS, which it doesn't. Maybe something like:
```suggestion
  Mbed TLS has the functionality that ``MBEDTLS_SSL_EXPORT_KEYS`` used to enable enabled by default, so uses of ``CONFIG_MBEDTLS_SSL_EXPORT_KEYS`` can simply be removed.
```

Author: Mmmm, I still prefer my version: it has the same meaning and less "enable" repetitions. But if you strongly suggest to change this, I can of course

Reviewer: It's fine. I just find the `assumed to be default on` possibly a bit misleading. Maybe just changing it to `assumed to be enabled` to at least remove the `default` would be more clear and okay for you?

  * ``CONFIG_MBEDTLS_ENTROPY_ENABLED`` is now :kconfig:option:``CONFIG_MBEDTLS_ENTROPY_C``,
  * ``CONFIG_MBEDTLS_ZEPHYR_ENTROPY`` is now :kconfig:option:``CONFIG_MBEDTLS_ENTROPY_POLL_ZEPHYR``.

* The Kconfig option ``CONFIG_MBEDTLS_SSL_EXPORT_KEYS`` was removed because the
  corresponding build symbol was removed in Mbed TLS 3.1.0 and is now assumed to

>   be enabled.
>   (:github:`77657`)

Reviewer: ```suggestion
  be enabled. (:github:`77657`)
```

Reviewer: Yeah it doesn't. Okay, didn't know there were other such, too bad but any way is fine for me!

>   corresponding build symbol was removed in Mbed TLS 3.1.0. Starting from this
>   release ``MBEDTLS_SSL_EXPORT_KEYS`` is assumed to be enabled in Mbed TLS.

Reviewer: Maybe this would be cleared:
```suggestion
  corresponding build symbol was removed in Mbed TLS 3.1.0 and is now assumed to be enabled.
```
Because now reading this, I thought that `Starting from this release` could refer to `this release` of Zephyr, not the Mbed TLs one you are referring to.
